packages/round-manager/src/context/__tests__/ApplicationContext.test.tsx

```diff
@@ -102,13 +103,100 @@ describe("<ApplicationProvider />", () => {
     });
   });
+
+  describe("useApplicationByRoundId()", () => {
+    it("provides applications based on given round id", async () => {
+      const expectedApplication = makeGrantApplicationData();
+      const expectedRoundId: string = expectedApplication.round!;
+
+      (getApplicationsByRoundId as any).mockResolvedValue([expectedApplication]);
+
+      render(
+        <ApplicationProvider>
+          <TestingUseApplicationByRoundIdComponent
+            expectedRoundId={expectedRoundId}
+          />
+        </ApplicationProvider>
+      );
+
+      expect(await screen.findByText(expectedRoundId)).toBeInTheDocument();
+    });
+
+    it("sets isLoading to true when getApplicationsByRoundId call is in progress", async () => {
+      const expectedApplication = makeGrantApplicationData();
+      const expectedRoundId: string = expectedApplication.round!;
+
+      (getApplicationsByRoundId as any).mockReturnValue(
+        new Promise<GrantApplication>(() => {})
+      );
+
+      render(
+        <ApplicationProvider>
+          <TestingUseApplicationByRoundIdComponent
+            expectedRoundId={expectedRoundId}
+          />
+        </ApplicationProvider>
+      );
+
+      expect(
+        await screen.findByTestId("is-loading-application-by-round-id")
+      ).toBeInTheDocument();
+    });
+
+    it("sets isLoading back to false and when getApplicationsByRoundId call succeeds", async () => {
+      const expectedApplication = makeGrantApplicationData();
+      const expectedRoundId: string = expectedApplication.round!;
+
+      (getApplicationsByRoundId as any).mockResolvedValue([expectedApplication]);
+
+      render(
+        <ApplicationProvider>
+          <TestingUseApplicationByRoundIdComponent
+            expectedRoundId={expectedRoundId}
+          />
+        </ApplicationProvider>
+      );
+
+      await waitFor(() => {
+        expect(
+          screen.queryByTestId("is-loading-application-by-round-id")
+        ).not.toBeInTheDocument();
+      });
+    });
+
+    it("sets isLoading back to false when getApplicationsByRoundId call fails", async () => {
+      const expectedApplication = makeGrantApplicationData();
+      const expectedRoundId: string = expectedApplication.round!;
+
+      (getApplicationsByRoundId as any).mockRejectedValue(new Error(":("));
+
+      render(
+        <ApplicationProvider>
+          <TestingUseApplicationByRoundIdComponent
+            expectedRoundId={expectedRoundId}
+          />
+        </ApplicationProvider>
+      );
+
+      await waitFor(() => {
+        expect(
+          screen.queryByTestId("is-loading-application-by-round-id")
+        ).not.toBeInTheDocument();
+      });
+
+      screen.getByTestId("application-by-round-id-error-msg");
+    });
+  });
 });
 
 const TestingUseApplicationByIdComponent = (props: {
-  expectedApplicationId?: string;
+  expectedApplicationId: string;
 }) => {
+
+  if (!props.expectedApplicationId) return(<></>);
```

Comment:
i'm seeing an `eslint` error here: `React Hook "useApplicationById" is called conditionally. React Hooks must be called in the exact same order in every component render`

do you have the husky hooks set up on your machine? you may need to run `yarn prepare` from the `packages/round-manager`, or run `husky install` at root

Reply:
also, maybe we should also add a `yarn lint` step to the CI checks